package cache;

//时装
message proDress{
    required uint32 dress_id = 1;       //时装编号
    required uint32 warrior_id = 2;     //适用弟子编号
    required uint32 level = 3;          //等级
}

//获得时装 返回proDressRes
message proGetDress{
    required uint32 user_id = 1;        //玩家编号
    required uint32 dress_id = 2;       //时装编号
}

//获得时装列表  返回proDressesRes
message proGetDresses{
    required uint32 user_id = 1;        //玩家编号
}

//根据阵容位置获得时装列表  返回proDresssRes
message proGetDressesByWarrior{
    required uint32 user_id = 1;        //玩家编号
    repeated uint32 warrior_list = 2;     //适用弟子编号
}

//添加时装  返回proDressRes
message proAddDress{
    required uint32 user_id = 1;        //玩家编号
    required proDress dress = 2;        //时装信息
}

//更新时装  返回proRes
message proUpdateDress{
    required uint32 user_id = 1;        //玩家编号
    required proDress dress = 2;        //时装信息
}

//删除时装 返回proRes
message proRemoveDress{
    required uint32 user_id = 1;        //玩家编号
    required uint32 dress_id = 2;       //时装编号
}

//包含时装的结果
message proDressRes{
    required uint32 code = 1;           //错误码
    optional proDress dress = 2;        //时装信息
}

//包含时装列表的结果
message proDressesRes{
    required uint32 code = 1;           //错误码
    repeated proDress dresses = 2;       //时装列表
}
